algolia/transport/transport.go

```diff
@@ -135,7 +135,9 @@ func (t *Transport) Request(
 			cancel()
 			return err
 		case Failure:
-			err = unmarshalToError(bodyRes)
+			if bodyRes != nil {
+				err = unmarshalToError(bodyRes)
+			}
```

Comment:
should we have a default error as fallback value here or are we sure we will return an actual `err` in case `bodyRes == nil`?

Reply:
Good point! Fixed ✔️